datasets/turkic_xwmt/turkic_xwmt.py

```diff
@@ -0,0 +1,132 @@
+"""
+
+_CITATION = """\
+@inproceedings{mirzakhalov2021large,
+  title={A Large-Scale Study of Machine Translation in Turkic Languages},
+  author={Mirzakhalov, Jamshidbek and Babu, Anoop and Ataman, Duygu and Kariev, Sherzod and Tyers, Francis and Abduraufov, Otabek and Hajili, Mammad and Ivanova, Sardana and Khaytbaev, Abror and Laverghetta Jr, Antonio and others},
+  booktitle={Proceedings of the 2021 Conference on Empirical Methods in Natural Language Processing},
+  pages={5876--5890},
+  year={2021}
+}
+"""
+
+_DATA_URL = (
+    "https://github.com/turkic-interlingua/til-mt/blob/6ac179350448895a63cc06fcfd1135882c8cc49b/xwmt/test.zip?raw=true"
+)
+
+
+class XWMTConfig(datasets.BuilderConfig):
+    """BuilderConfig for XWMT."""
+
+    def __init__(self, language_pair=(None, None), **kwargs):
+        """BuilderConfig for XWMT.
+
+        Args:
+            for the `datasets.features.text.TextEncoder` used for the features feature.
+          language_pair: pair of languages that will be used for translation. Should
+            contain 2-letter coded strings.
+          **kwargs: keyword arguments forwarded to super.
+        """
+        name = "%s-%s" % (language_pair[0], language_pair[1])
+
+        description = ("Translation dataset from %s to %s") % (language_pair[0], language_pair[1])
+        super(XWMTConfig, self).__init__(
+            name=name,
+            description=description,
+            version=datasets.Version("1.1.0", ""),
+            **kwargs,
+        )
+
+        # Validate language pair.
+        source, target = language_pair
+
+        assert source in _LANGUAGES, ("Config language pair must be one of the supported languages, got: %s", source)
+        assert target in _LANGUAGES, ("Config language pair must be one of the supported languages, got: %s", source)
+
+        self.language_pair = language_pair
+
+
+class TurkicXWMT(datasets.GeneratorBasedBuilder):
+    """XWMT machine translation dataset."""
+
+    BUILDER_CONFIGS = [
+        XWMTConfig(
+            language_pair=(lang1, lang2),
+        )
+        for lang1 in _LANGUAGES
+        for lang2 in _LANGUAGES
+        if lang1 != lang2
+    ]
+
+    def _info(self):
+        source, target = self.config.language_pair
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {"translation": datasets.features.Translation(languages=self.config.language_pair)}
+            ),
+            supervised_keys=(source, target),
+            homepage="https://github.com/turkicinterlingua/til-mt",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        path = dl_manager.download_and_extract(_DATA_URL)
+
+        source, target = self.config.language_pair
+        source_path = f"{path}/test/{source}-{target}/{source}-{target}.{source}.txt"
+        target_path = f"{path}/test/{source}-{target}/{source}-{target}.{target}.txt"
```

Comment:
Please use `os.path.join` or you may have issues on windows or in streaming mode:
```suggestion
        source_path = os.path.join(path}, "test", f"{source}-{target}", f"{source}-{target}.{source}.txt")
        target_path = os.path.join(path}, "test", f"{source}-{target}", f"{source}-{target}.{target}.txt")
```